src/LuaPrinter.ts

```diff
@@ -82,11 +110,27 @@ export class LuaPrinter {
         return this.currentIndent + input;
     }
 
-    private printBlock(block: tstl.Block): string {
-        return this.ignoreDeadStatements(block.statements).map(s => this.printStatement(s)).join("");
+    private createSourceNode(
```

Comment:
After fiddling around with this a little I found that defining this function as follows leads to much nicer code:
```ts
private createSourceNode(node: tstl.Node, chunks: Array<(string | SourceNode)>): SourceNode {
    const line = node.line !== undefined ? node.line + 1 : undefined;
    const column = node.column !== undefined ? node.column + 1 : undefined;

    return new SourceNode(line, column, this.sourceFile, chunks);
}
```